app/code/Magento/Indexer/Model/ProcessManager.php

```diff
@@ -135,11 +148,20 @@ private function isSetupMode(): bool
      */
     private function startChildProcess(callable $userFunction)
     {
-        // phpcs:ignore Magento2.Functions.DiscouragedFunction
-        $status = call_user_func($userFunction);
-        $status = is_int($status) ? $status : 0;
-        // phpcs:ignore Magento2.Security.LanguageConstruct.ExitUsage
-        exit($status);
+        try {
+            // phpcs:ignore Magento2.Functions.DiscouragedFunction
+            $status = call_user_func($userFunction);
+            $status = is_int($status) ? $status : 0;
+        } catch (\Throwable $e) {
+            $status = 1;
+            $this->logger->error(
+                __('Child process failed with message: %1', $e->getMessage()),
+                $e->getTrace()
+            );
```

Comment:
Just found that 2nd argument should be array, also sending whole stack trace as an array is quite not useful thing. It's better to use something like this
```suggestion
            $this->logger->error(
                __('Child process failed with message: %1', $e->getMessage()),
                ['exception' => (string)$e]
            );
```

Or 
```suggestion
            $this->logger->error(
                __('Child process failed with message: %1', $e->getMessage()),
                ['exception_trace' => $e->getTraceAsString()]
            );
```